job/server/src/main/java/alluxio/master/job/tracker/CmdJobTracker.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.job.tracker;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileSystemContext;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.JobDoesNotExistException;
+import alluxio.job.CmdConfig;
+import alluxio.job.cmd.load.LoadCliConfig;
+import alluxio.job.cmd.migrate.MigrateCliConfig;
+import alluxio.job.cmd.persist.PersistCmdConfig;
+import alluxio.job.wire.Status;
+import alluxio.master.job.JobMaster;
+import alluxio.master.job.common.CmdInfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * CmdJobTracker to schedule a Cmd job to run.
+ */
+@ThreadSafe
+public class CmdJobTracker {
```

Comment:
since this code runs in the job master, can we quantify how much additional heap memory is required for job state management?  for example, before and after to load 100k files, what the difference would be?

Reply:
Will do this test in this week to quantify.